arroyo-sql/src/lib.rs

```diff
@@ -164,6 +175,83 @@ impl ArroyoSchemaProvider {
+        &self,
+        query: &Statement,
+    ) -> Result<(String, Vec<StructField>, Operator)> {
+        let Statement::CreateTable {
+        name,
+        columns,
+        with_options,
+        ..
+    } = query else {
+        bail!("expected create table statement");
+    };
+        let connection_name = with_options
+            .iter()
+            .filter_map(|option| match option.name.value.as_str() {
+                "connection" => Some(value_to_inner_string(&option.value)),
+                _ => None,
+            })
+            .next()
+            .ok_or_else(|| anyhow!("expect connection in with block"))??;
+        let connection = self
+            .connections
+            .get(&connection_name)
+            .ok_or_else(|| anyhow!("connection {} not found", connection_name))?;
+
+        let json_fields: Vec<StructField> = columns
+            .iter()
+            .map(|column| {
+                let name = column.name.value.to_string();
+                let data_type = convert_data_type(&column.data_type)?;
+                let nullable = !column
+                    .options
+                    .iter()
+                    .any(|option| matches!(option.option, ColumnOption::NotNull));
+                Ok(StructField {
+                    name,
+                    alias: None,
+                    data_type: TypeDef::DataType(data_type, nullable),
+                })
+            })
+            .collect::<Result<Vec<_>>>()?;
+
+        match connection.connection_type {
+            Some(ConnectionType::Kafka(ref kafka_connection)) => {
+                let topic: String = with_options
+                    .iter()
+                    .filter_map(|option| match option.name.value.as_str() {
+                        "topic" => Some(value_to_inner_string(&option.value)),
+                        _ => None,
+                    })
+                    .next()
+                    .ok_or_else(|| anyhow!("expect topic in with block"))??;
+                Ok((
+                    name.to_string(),
+                    json_fields,
+                    Operator::KafkaSource {
+                        topic,
+                        // split to a vec
+                        bootstrap_servers: kafka_connection
+                            .bootstrap_servers
+                            .split(",")
+                            .map(|s| s.to_string())
+                            .collect(),
+                        offset_mode: OffsetMode::Latest,
+                        kafka_input_format: SerializationMode::Json,
+                        // TODO: use setting for account
+                        messages_per_second: 10000000,
+                        client_configs: auth_config_to_hashmap(
+                            kafka_connection.auth_config.clone(),
+                        ),
+                    },
+                ))
+            }
+            _ => bail!("unsupported connection type"),
```

Comment:
I think it's worth being explicit here so that we know where we need to update when add new connection types. Additionally, this case is actually `None`, not an unsupported connection type, so this actually means that the protobuf is malformed somehow

Reply:
Could also be Kinesis, cleaned up the errors.